lib/services.js

```diff
@@ -245,79 +262,80 @@ export default {
         log.trace('object metadata', { omVal });
         // If this is not the completion of a multipart upload
         // parse the headers to get the ACL's if any
-        if (!multipart) {
-            const parseAclParams = {
-                headers,
-                resourceType: 'object',
-                acl: omVal.acl,
-                log,
-            };
-            log.trace('parsing acl from headers');
-            acl.parseAclFromHeaders(parseAclParams, (err, parsedACL) => {
-                if (err) {
-                    log.debug('error parsing acl', { error: err });
-                    return cb(err);
+        async.waterfall([
+            callback => {
+                if (multipart) {
+                    return callback();
                 }
-                omVal.acl = parsedACL;
-                metadata.putObjectMD(bucketName, objectKey, omVal, {}, log,
-                err => {
+                const parseAclParams = {
+                    headers,
+                    resourceType: 'object',
+                    acl: omVal.acl,
+                    log,
+                };
+                log.trace('parsing acl from headers');
+                acl.parseAclFromHeaders(parseAclParams, (err, parsedACL) => {
                     if (err) {
-                        log.error('error from metadata', { error: err });
-                        return cb(err);
+                        log.warn('error parsing acl', { error: err });
+                        return callback(err);
                     }
-                    log.trace('object successfully stored in metadata');
-                    return cb(err, contentMD5);
+                    omVal.acl = parsedACL;
+                    return callback();
                 });
-                return undefined;
-            });
-        } else {
-            metadata.putObjectMD(bucketName, objectKey, omVal, {}, log, err => {
-                if (err) {
-                    log.error('error from metadata', { error: err });
-                    return cb(err);
+                return null;
+            },
+            callback => metadata.putObjectMD(bucketName, objectKey, omVal,
+                    options, log, callback),
+        ], (err, data) => {
+            if (err) {
+                log.error('error from metadata', { error: err });
+                return cb(err);
+            }
+            log.trace('object successfully stored in metadata');
+            // for versioning only, other features need to process their data
```

Comment:
perhaps: // if versioning is enabled, data will be returned from metadata containing a versionId which some APIs will need sent back to it